src/spikeinterface/preprocessing/filter.py

```diff
@@ -146,10 +150,16 @@ def get_traces(self, start_frame, end_frame, channel_indices):
         import scipy.signal
 
         if self.filter_mode == "sos":
-            filtered_traces = scipy.signal.sosfiltfilt(self.coeff, traces_chunk, axis=0)
+            if causal:
+                filtered_traces = np.flip(scipy.signal.sosfilt(self.coeff, np.flip(traces_chunk)))
+            else:
+                filtered_traces = scipy.signal.sosfiltfilt(self.coeff, traces_chunk, axis=0)
         elif self.filter_mode == "ba":
             b, a = self.coeff
-            filtered_traces = scipy.signal.filtfilt(b, a, traces_chunk, axis=0)
+            if causal:
+                filtered_traces = np.flip(scipy.signal.lfilter(b, a, np.flip(traces_chunk), axis=0))
```

Comment:
This and the same comment for line 154 is unecessarily fliping the channels as well. I think you should only flip the time axis.

Reply:
Hi, yes. You are right about the channels flipping as well. that is missing: the correct line should look like this, right?: 
```
filtered_traces = np.flip(scipy.signal.lfilter(b, a, np.flip(traces_chunk, axis = 0), axis=0))
```

Yes, BackwardsFilter sounds better and it is more self explanatory.